### PR TITLE
Ban

### PR DESCRIPTION
**Pull request by y1bs**
# Feature
B.3.1. View Banned Student List
B.3.4 View Banned Sensei List
B.3.12 Ban Student
B.3.13 Ban Sensei
B.3.14 Unban Student
B.3.15 Unban Sensei

# Changelog:

# Merge/Review Order

These other PRs should be merged or reviewed first

This PR blocks these other PRs

## Checklist:

- [ ] Merged latest develop
- [ ] PR title makes sense

### Banned Sensei List
image

### Sensei Profile
image

### Pop Confirm
image

### CODE REVIEW

File: src/services/admin/index.js [hunk 1]
```diff
@@ -225,11 +213,46 @@ export async function getAllMentorshipContracts() {
...
     .then(response => {
-      // console.log('Here', response)
       if (response && !isNil(response.data)) {
         if (response.data.success) return response.data.mentorshipContracts
       }
       return false
     })
     .catch(err => console.log(err))
 }
+
```
Comment (rawrrencee) [inline src/services/admin/index.js:223]: paiseh, the getAllBannedStudent(s) and getAllBannedSensei(s) can add an 's' for the plural HAHA psps

File: src/components/Admin/UsersManagement/SenseiProfileComponent/index.js [hunk 2]
```diff
@@ -167,20 +163,40 @@ const SenseiProfileComponent = () => {
...
-            type="primary"
-            size="large"
-            shape="round"
-            onClick={onBack}
-            icon={<ArrowLeftOutlined />}
+          <BackBtn />
+        </div>
+        <div className="col-12 col-md-auto col-lg-auto mt-4 mt-md-0 text-center text-md-right">
+          <Popconfirm
+            title="Do you wish to change account status?"
```
Comment (rawrrencee) [inline src/components/Admin/UsersManagement/SenseiProfileComponent/index.js]: A bit mafan but can apply the same check for sensei.status === ACTIVE and show 
"Do you wish to ban this user?" 
"Do you wish to unban this user?"

**Review by rawrrencee — DISMISSED** [review]
no issues other than the 2 minor commented ones! thanks! approved unless changes need to be made

File: src/services/admin/index.js (hunk 1 above)
Comment (y1bs) [reply src/services/admin/index.js:223]: Cann

File: src/components/Admin/UsersManagement/SenseiProfileComponent/index.js (hunk 2 above)
Comment (y1bs) [reply src/components/Admin/UsersManagement/SenseiProfileComponent/index.js]: Okies, I changed to do you wish to ban this student/sensei depending on the user type, you ok with that?

Comment (rawrrencee) [reply src/components/Admin/UsersManagement/SenseiProfileComponent/index.js]: oh ok! then unban also same right? haha

Comment (y1bs) [reply src/components/Admin/UsersManagement/SenseiProfileComponent/index.js]: Yeps correct hahaha, so got 2 popconfirms per profilecomponent. One for ban, one for unban

File: src/components/Admin/UsersManagement/SenseiProfileComponent/index.js [hunk 3]
```diff
@@ -181,22 +181,29 @@ const SenseiProfileComponent = () => {
...
+              okType="danger"
+            >
               <Button danger block shape="round" size="large" icon={<StopOutlined />}>
                 Ban Account
               </Button>
-            ) : (
+            </Popconfirm>
+          ) : (
+            <Popconfirm
+              title="Do you wish to ban this sensei?"
```
Comment (rawrrencee) [inline src/components/Admin/UsersManagement/SenseiProfileComponent/index.js]: is this supposed to be unban ah! psps

Comment (y1bs) [reply src/components/Admin/UsersManagement/SenseiProfileComponent/index.js]: Eh yeah sorry i blind XD

Comment (rawrrencee) [reply src/components/Admin/UsersManagement/SenseiProfileComponent/index.js]: no la!!! too much codeeee

**Review by rawrrencee — APPROVED** [review]
LGTM (i had to google what it meant the first time you used it @y1bs ...) AHAHHA

**Comment by y1bs** [comment]
> LGTM (i had to google what it meant the first time you used it @y1bs ...) AHAHHA

I learnt it from here! XD https://www.youtube.com/watch?v=rR4n-0KYeKQ&ab_channel=JomaTech

**Pull request merged**